settings.py

```diff
@@ -1056,6 +1056,16 @@
 ]
 COMMCARE_EXTENSIONS = []
 
+# Provides extensions with ability to customize form processing per domain.
+# For each domain provide a list of pre-processors that would be instantiated
+# and are passed the form xml for processing
+XFORM_PRE_PROCESSORS = {}
```

Comment:
this should be an extension point in `corehq.form_processor.extension_points`:

```python
@extension_point(result_format=ResultFormat.FIRST)
def get_form_submission_context_class():
    """docs"""
```

Then ICDS can extend as follows:
```python
@get_form_submission_context_class.extend()
def icds_form_submission_context():
    return ICDSFormSubmissionContext
```

Reply:
implemented in https://github.com/dimagi/commcare-hq/pull/28409/commits/3578ca3a5ac8239e474d9b77171b6c792577b2a1